comment/managers/blocker.py

```diff
@@ -0,0 +1,21 @@
+from django.db import models
+
+
+class BlockedUserManager(models.Manager):
+    def is_user_blocked(self, user_id=None, email=None):
+        try:
+            user_id = int(user_id)
+            return self.filter(user_id=user_id, blocked=True).exists()
+        except (ValueError, TypeError):
+            if not email:
+                return False
+            return self.filter(email=email, blocked=True).exists()
```

Comment:
i feel this function can be separated in two different ones, even though function definition tells you that both parameters are not required, at least one of them is. I don't find much use of calling this without any parameters.

Reply:
suggestion?

Reply:
maybe, something along these lines:
```suggestion
def is_user_blocked_by_email(self, email):
    return self.filter(email=email, blocked=True).exists()
    
def is_user_blocked_by_id(self, user_id):
    return self.filter(user_id=user_id, blocked=True).exists()
```

Reply:
```python
    def is_user_blocked(self, user_id=None, email=None):
        if user_id:
            return self.is_user_blocked_by_id(user_id)
        elif email:
            return self.is_email_blocked(email)
        raise TypeError('user_id and email cannot be both None')
```

Reply:
why would you want to raise a typeerror, when this can be generated directly by the interpreter when using a restricted function signature, like the one i suggested above.

Reply:
if you really want to do both of this in one function:
```python
def is_user_blocked(self, user_id=None, email=None):
    if not email and not user:
         raise
    if user_id:
         # do your thing
     elif:
        # do your thing
```
although python suggests the philosophy to ask for forgiveness rather than ask for permission, in this case, we won't have to do the database transaction.

although,  i'm still not convince whether writing a function signature that suggests there are two optional parameters but both of them aren't exclusively optional, is a good design strategy.

Reply:
Becasue I want to keep the logic inside the manager so that I can always use one method to check if the user blocked no matter passing the email or user id. Otherwise I need to make this checks in every place I need to check if the user is blocked. see this https://github.com/Radi85/Comment/pull/201/files#diff-150a36628526769f378f79af3ad910fb6d2e0d6c6ddf28c8cc3a20c8f5d8ba63R34 and https://github.com/Radi85/Comment/pull/201/files#diff-a3711352663d6d29db96071e9efed0a4c0c9d556402c29d56111fcc47bda266dR43

Reply:
> although, i'm still not convince whether writing a function signature that suggests there are two optional parameters but both of them aren't exclusively optional, is a good design strategy.

The original implementaion will just return False in case of not passing any params and I believe I would keep this implementation.

```python
def is_user_blocked(self, user_id=None, email=None):
        if user_id:
            return self.is_user_blocked_by_id(user_id)
        elif email:
            return self.is_email_blocked(email)
        return False
```

Reply:
so, here when both `user_id` and `email` are none, you can just return `False`, without having to do the queries to the database, which will be more efficient I think.